### PR TITLE
[FIXED] Bug on single replica restore would deliver duplicate messages.

### PR DESCRIPTION
**Pull request by derekcollison**
Fixed a bug that on restore of single streams in clustered mode would subscribe to the stream subjects twice.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -854,6 +860,7 @@ func (mset *Stream) unsubscribeToStream() error {
 	for _, subject := range mset.config.Subjects {
 		mset.unsubscribeInternal(subject)
 	}
+	mset.active = true
```
Comment (kozlovic) [inline server/stream.go:863]: Should it be false here???

File: test/jetstream_cluster_test.go [hunk 2]
```diff
@@ -2612,6 +2612,53 @@ func TestJetStreamRestartAdvisories(t *testing.T) {
...
+
+	sub, err := js.SubscribeSync("foo", nats.Durable("dlc"))
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	js.Publish("foo", []byte("msg1"))
+	if _, err := sub.NextMsg(time.Second); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
```
Comment (kozlovic) [inline test/jetstream_cluster_test.go:2638]: The message is not ack'ed and therefore could be redelivered. Should we ack it and wait for ack processing before continuing?

File: test/jetstream_cluster_test.go [hunk 3]
```diff
@@ -2612,6 +2612,53 @@ func TestJetStreamRestartAdvisories(t *testing.T) {
...
+
+	// Send second msg
+	js.Publish("foo", []byte("msg2"))
+	msg, err := sub.NextMsg(time.Second)
+	if err != nil {
+		t.Fatalf("Error getting message: %v", err)
+	}
+	if string(msg.Data) != "msg2" {
+		t.Fatalf("Unexpected message: %s", msg.Data)
+	}
```
Comment (kozlovic) [inline test/jetstream_cluster_test.go:2653]: Same here, we should ack here?

File: test/jetstream_cluster_test.go (hunk 2 above)
Comment (derekcollison) [reply test/jetstream_cluster_test.go:2638]: I had that in the test but default will never have us see it in this test so removed. Could add back though..

File: test/jetstream_cluster_test.go (hunk 3 above)
Comment (derekcollison) [reply test/jetstream_cluster_test.go:2653]: Fixed. Thanks.

File: test/jetstream_cluster_test.go (hunk 2 above)
Comment (derekcollison) [reply test/jetstream_cluster_test.go:2640]: ok added Akc() back.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**